jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -285,7 +301,74 @@ def search(
 
                 query_docs[doc_idx].matches.append(match)
 
-    def _train(self, index, data: 'np.ndarray', *args, **kwargs) -> None:
+    @requests(on='/train')
```

Comment:
Do I get the following right:

We would have a TrainingFlow running all the time (or when training is needed), which is able to react an `train` requests. It will train a Faiss index, fill it with the latest data and dump it somewhere (potentially to S3 or so). Then a running FaissSearcher can be restarted with the new dump path and after loading the index directly start serving data.

I like it. My first reaction was "That is a miss-use of the Jina Flow concept!! How dare you!", but actually it has quite some charm. :)